annotation/tests/test_tasks_services.py

```diff
@@ -1472,3 +1472,201 @@ def test_get_common_ids(
+        ((), {}, {}, [], []),
+    ],
+)
+def test_get_links_and_children(
+    common_objs_ids: Tuple[int],
+    all_tasks_objs: Dict[int, Dict[str, List[Dict[str, int]]]],
+    id_mapping: Dict[Tuple[int], int],
+    expected_links: List[List[Dict[str, int]]],
+    expected_children: List[List[int]],
+):
+    with patch("annotation.tasks.services.get_new_id") as mock_get_new_id:
+        mock_get_new_id.side_effect = lambda old_id, _: (
+            old_id if old_id in [2, 3] else None
+        )
+        result_links, result_children = services.get_links_and_children(
+            common_objs_ids, all_tasks_objs, id_mapping
+        )
+        assert result_links == expected_links
+        assert result_children == expected_children
+
+
+@pytest.mark.parametrize(
+    ("items", "expected"),
+    [
+        (
+            [
+                [{"a": 1, "b": 2, "c": 3}, {"d": 4}, {"e": 5}],
+                [{"d": 4}, {"f": 6}],
+                [{"d": 4}, {"g": 7}],
+            ],
+            [{"d": 4}],
+        ),
+        (
+            [
+                [{"a": 1}],
+                [],
+                [{"b": 2}],
+            ],
+            [],
+        ),
+        (
+            [
+                [{"a": 1}, {"b": 2}],
+            ],
+            [{"a": 1}, {"b": 2}],
+        ),
+        ([], []),
+    ],
+)
+def test_get_common_values(
+    items: List[List[Dict[str, int]]],
+    expected: List[Dict[str, int]],
+):
+    assert services.get_common_values(items) == expected
+
+
+@pytest.mark.parametrize(
+    (
+        "all_tasks_objs",
+        "doc_objs",
+        "id_mapping",
+        "expected_ids",
+        "expected_links",
+        "expected_id",
+        "expected_children",
+    ),
+    [
+        (
+            {
+                "task1": {"objects": [{"id": 1}, {"id": 2}]},
+                "task2": {"objects": [{"id": 3}, {"id": 4}]},
+            },
+            {
+                "doc1": {
+                    "objects": [
+                        {"id": 1, "links": [2], "children": [3]},
+                        {"id": 2},
+                        {"id": 3},
+                    ]
+                }
+            },
+            {(1,): 101, (2,): 102, (3,): 103},
+            [101, 102, 103],
+            [101],
+            [3, 1],
+            [3],
+        ),
+        (
+            {
+                "task1": {"objects": [{"id": 1}, {"id": 2}, {"id": 3}]},
+                "task2": {"objects": [{"id": 4}, {"id": 5}, {"id": 6}]},
+            },
+            {
+                "doc1": {
+                    "objects": [
+                        {"id": 1, "links": [2], "children": [3, 4]},
+                        {"id": 2},
+                        {"id": 3},
+                    ]
+                }
+            },
+            {(1,): 101, (2,): 102, (3,): 103, (4,): 104},
+            [101, 102, 103],
+            [101],
+            [3, 1],
+            [101],
+        ),
+    ],
+)
+def test_change_ids(
```

Comment:
this test is huge, could you simplify it? it is not mandatory to test everything in a single test